pkg/matcher/expr.go

```diff
@@ -0,0 +1,52 @@
+package matcher
+
+import "fmt"
+
+type (
+	// SimpleExpr is a simple expression to describe the condition:
+	//     (includes[0].Match(v) || includes[1].Match(v) || ...) && !(excludes[0].Match(v) || excludes[1].Match(v) || ...)
+	SimpleExpr struct {
+		Includes []string `yaml:"includes" json:"includes"`
+		Excludes []string `yaml:"excludes" json:"excludes"`
+
+		matcher Matcher
+	}
+)
+
+// Parse parse the given matchers in Includes and Excludes
+func (s *SimpleExpr) Parse() error {
+	var (
+		includes = FALSE()
+		excludes = FALSE()
+	)
+	for _, item := range s.Includes {
+		m, err := Parse(item)
+		if err != nil {
+			return fmt.Errorf("parse matcher %q error: %v", item, err)
+		}
+		includes = Or(includes, m)
+	}
+	for _, item := range s.Excludes {
+		m, err := Parse(item)
+		if err != nil {
+			return fmt.Errorf("parse matcher %q error: %v", item, err)
+		}
+		excludes = Or(excludes, m)
+	}
+
+	if len(s.Includes) == 0 {
+		includes = TRUE()
+	}
```

Comment:
if `Includes` and `Excludes` are empty matcher should return `false`, not `true`

Reply:
if Includes is empty, we should default them to `true` or empty filter config will lead to exclude everything.

Reply:
Empty config == don't use matcher.

Otherwise we need to add bool param to every filer.

https://github.com/netdata/go.d.plugin/blob/a212685749537aa85b294842d90f9f26b1faaee0/modules/coredns/collect.go#L48-L53

like `if cd.DoPerServerStats { ... }`

Reply:
for example:
```
filter:
  excludes:
  - = foo
```
only `foo` should be excluded, others should be pass.

if you set empty `includes` false, everything will be excluded.

Reply:
or add `IsEmpty` or something method to it, to know when to use it

Reply:
`if filter == TRUE() {...}` is not enough?

Reply:
Enough, but please make default (empty)`FALSE`, because it is logical - no matchers no match